aspects/aspects.go

```diff
@@ -444,9 +444,17 @@ type expandedMatch struct {
 	value interface{}
 }
 
+// maxValueDepth is the limit on a value's nestedness. Since creating a highly
+// nested JSON value takes only few bytes per level but makes snapd recurse
+// creating a 2Kb stack, we need to prevent this from being abused.
```

Comment:
How did we arrive at 2Kb, is this worth checking/testing?

Reply:
It's taken from the source https://go.dev/src/runtime/stack.go